### DISCUSSION TITLE
Dayspring Theme

### DISCUSSION THREAD

**Discussion by mettaseva** [oleeskild/obsidian-digital-garden#98] [Q&A] [answered]
Earlier this afternoon I reset my digital garden to use the Dayspring theme. (FYI, the theme settings in the my desktop app match the plugin settings.)

However, this new theme is _not_ displaying in my published site in spite of the fact that I've published multiple edits to my site, and I've cleared my Brave browser cache several times.

Where/how should I begin troubleshooting this issue?

LATER: Just now discovered my site edits are not syncing now. I've tried syncing individual pages as well as using the publication center. 
Please advise.

With my thanks in advance,
Metta

**Comment by amperag** [comment]
Hey @mettaseva, I think you've got the same problem as I had: #97.
Ole pushed a new version of digital garden (2.23.1) which fixed the problem for me (see below).
Basically you need to (1) update your digital garden plugin in Obsidian and then (2) update your site settings again and it should work.

---
```
So the issue indeed was the URL of the theme. The plugin assumed that the branch for the theme was always named master. This was not the case with the Wasp theme, which used "main" instead. I've changed the plugin to instead use whatever the theme has set as their default branch. 
If you update the plugin, and switch to a new theme, before switching back to Wasp again to make sure the plugin uses the new URL, it should work again. Let me know if that fixed it 😄
```

_Originally posted by @oleeskild in https://github.com/oleeskild/obsidian-digital-garden/issues/97#issuecomment-1312531128_

**Reply by oleeskild** [reply]
I thought so as well at first, but this was actually a slightly different problem, where themes for Obsidian version 1.0 are using different conventions for naming their theme files. I've updated the plugin with a new version that should handle this case as well, so that both legacy themes, and newer themes are supported.

**Reply by mettaseva** [reply]
@amperag ~ Thank you very much for taking time to weigh in on this. Your help and feedback is appreciated, even if the problem I reported was related to another issue. :-)

**Comment by oleeskild** [comment, marked as answer]
Hi @mettaseva!
I've taken a closer look at how themes are handled, and the problem is that obsidian changed some parts of how they handled themes after the release of obsidian version 1.0. 

Long story short: the plugin only supported old themes. The dayspring theme is only available for version 1.0 and up. I've just published version 1.23.2, which should fix that issue. 

So your issue should be fixed if you update the plugin, and switch to a new theme, before switching back to Dayspring again to make sure the plugin uses the latest version. Let me know if that fixes it.

**Comment by mettaseva** [comment]
Thank you very much, @oleeskild, for your prompt and helpful follow-up! I appreciate you jumping on this issue so quickly and providing such a fast turn-around on the plugin update.

I followed your recommended steps: update plugin, change to another theme, change theme back to Dayspring.

However, it looks like I'm still having a more fundamental problem with my site sync. As I reported earlier, _none_ of my text edits are now syncing, so I'm afraid the other plugin changes may not be syncing either.

Any thoughts or suggestions about how I should begin troubleshooting this sync issue ~ or do I need to work this out through Netlify?

With my thanks again,
Metta

**Reply by oleeskild** [reply]
I think the syncing issues is because of the theme. When the site is building in Netlify, it looks for the theme file to download it. When that fails, the whole building process fails, and so none of your changes will show.

Looking in your [digital garden repository](https://github.com/mettaseva/digitalgarden), I suspect that maybe you forgot to click the "Apply Settings" button when changing the themes. Could that be it? 

![CleanShot 2022-11-13 at 09 37 59@2x](https://user-images.githubusercontent.com/6201338/201513205-1a00260f-8c66-499c-bc8a-b1b39e9a8198.png)

**Reply by mettaseva** [reply]
That did the trick! :-) Thanks so much for your prompt and helpful feedback.
I was making all my theme changes in the main vault, and missed the fact that I had to update them _within_ the plugin, as well. 
Appreciate your detective work in figuring out what the issue was!
